main.js

```diff
@@ -0,0 +1,386 @@
+  if (typeof minuend !== "object" || typeof subtrahend !== "object") {
+    throw "setDifference needs objects as arguments";
+  }
+  let returnArray = [];
+  let flag = 0; //used to mark if two values in the arrays are similar at any point
+  for (i in minuend) {
+    for (j in subtrahend) {
+      if (minuend[i] === subtrahend[j]) {
+        flag = 1;
+        break;
+      }
+    }
+    if (flag === 0) {
+      returnArray.push(minuend[i]);
+    }
+    flag = 0;
+  }
+  return returnArray;
+}
+
+function randomInteger(min, max) {
+  let range = max - min + 1;
+  return min + Math.floor(Math.random() * range);
+}
+
+function XOToPlayerName(whoseTurn) {
+  if (whoseTurn === "X") {
+    return playerX.name;
+  }
+  if (whoseTurn === "O") {
+    return playerO.name;
+  }
+}
+
+function XOToPlayerType(whoseTurn) {
+  if (whoseTurn === "X") {
+    return playerX.type;
+  }
+  if (whoseTurn === "O") {
+    return playerO.type;
+  }
+}
+
+function resetTimer() {
+  startTime = Date.now();
+}
+
+function runTimer() {
+  setTimeout(() => {
+    if (currentState === "gameStarted") {
+      let totalMs = Date.now() - startTime; //I continually reference date.now because if I just waited 1000ms and then added 1 totalSeconds, the inaccuracy of setTimeout would eventually become non-negligible
+      let totalSeconds = Math.round(totalMs / 1000);
+      let seconds = totalSeconds - 60 * Math.floor(totalSeconds / 60);
+      let totalMinutes = Math.floor(totalSeconds / 60);
+      let minutes = totalMinutes - 60 * Math.floor(totalMinutes / 60);
+      let hours = Math.floor(totalMinutes / 60);
+      if (seconds < 10) {
+        seconds = "0" + seconds;
+      }
+      if (minutes < 10) {
+        minutes = "0" + minutes;
+      }
+      if (hours < 10) {
+        hours = "0" + hours;
+      }
+      timer.textContent =
+        "Time elapsed: " + hours + ":" + minutes + ":" + seconds;
+      runTimer();
+    }
+  }, 100);
+}
+runTimer();
+
+function computerPlays() {
+  if (XOToPlayerType(whoseTurn) === "Computer") {
+    let allCells = Array(9);
+    let i = 0;
+    while (i < 9) {
+      allCells[i] = i;
+      i++;
+    }
+    let unoccupiedCells = setDifference(allCells, occupiedCells.all);
+    console.log("these cells are unoccupied: " + unoccupiedCells);
+    let choiceNum =
+      unoccupiedCells[randomInteger(0, unoccupiedCells.length - 1)];
+    console.log("computer: I have chosen cell number " + choiceNum);
+    let choiceCell = document.getElementById("cell-" + choiceNum);
+    playCell(choiceCell);
+  }
+}
+
+let optionsButton = document.getElementById("options");
+let optionsDialog = document.getElementById("optionsDialog");
+let selectX = document.getElementsByTagName("select")[0];
+let selectO = document.getElementsByTagName("select")[1];
+let confirmButton = document.getElementById("confirmBtn");
```

Comment:
Another organization best practice is to have all your global variables defined at the top of the file.  This makes sure that variables are defined before you use them, and helps prevent duplicate names which will break your code.